mewarpx/mewarpx/plotting.py

```diff
@@ -0,0 +1,633 @@
+        raise ValueError("Unrecognized dimension dim = {}".format(dim))
+    if dim == 1:
+        return[axis_dict_2d['z'], axis_dict_2d['x'], None, None]
+    for ii, axis in enumerate(axes):
+        if dim == 2 and axis not in axis_labels_2d:
+            raise ValueError("Unrecognized axis {} for 2D".format(axis))
+        if dim == 3 and axis not in axis_labels_3d:
+            if axis == 'r':
+                axes[ii] = 'x'
+            else:
+                raise ValueError("Unrecognized axis {} for 3D".format(axis))
+    if axes[0] == axes[1] or (axes[0] in ['r', 'x']
+                              and axes[1] in ['r', 'x']):
+        raise ValueError("axis1 and axis2 must be different")
+    if dim == 2:
+        return [axis_dict_2d[axes[0]], axis_dict_2d[axes[1]], None, None]
+    xaxis = axis_dict_3d[axes[0]]
+    yaxis = axis_dict_3d[axes[1]]
+    sliceaxis = (set((0, 1, 2)) - set((xaxis, yaxis))).pop()
+    s_str = (set(('x', 'y', 'z')) - set((axes[0], axes[1]))).pop()
+    return [xaxis, yaxis, sliceaxis, s_str]
+
+def get_2D_field_slice(data, xaxis, yaxis, slicevec=None, slicepos=None):
+    """Return appropriate 2D field slice given the geometry.
+    Arguments:
+        data (np.ndarray): 2D or 3D array, depending on geometry
+        xaxis (int): Index of abscissa dimension of data
+        yaxis (int): Index of ordinate dimension of data
+        sliceaxis (int): Index of dimension of data to slice from. None for 2D.
+        slicevec (np.ndarray): 1D vector of positions along slice. None for 2D,
+            or to take middle element in 3D.
+        slicepos (float): Position to slice along sliceaxis (m). Default 0 if
+            slicevec != None; ignored if slicevec == None.
+    Returns:
+        slice (np.ndarray): 2D array. Ordinate is the first dimension of the
+        array, abscissa the 2nd.
+    """
+    data = np.array(data)
+    dim = len(data.shape)
+    if dim == 1:
+        data = np.tile(data, (2, 1))
+        # Flip x and y?
+        if xaxis < yaxis:
+            return data.T
+        return data
+    if dim == 2:
+        # if slicevec is not None or slicepos is not None:
+        #      logger.warning("slicevec and slicepos ignored for 2D data in "
+        #                     "get_2D_field_slice()")
+        # Flip x and y?
+        if xaxis < yaxis:
+            return data.T
+        return data
+    sliceaxis = (set((0, 1, 2)) - set((xaxis, yaxis))).pop()
+    if slicevec is None:
+        # if slicepos is not None:
+        #     logger.warning("slicepos ignored when slicevec == None in "
+        #                    "get_2D_field_slice()")
+        idx = data.shape[sliceaxis] // 2
+    else:
+        if slicepos is None:
+            slicepos = 0.0
+        idx = np.argmin(np.abs(slicevec - slicepos))
+    dslice = data.take(idx, axis=sliceaxis)
+    if xaxis < yaxis:
+        return dslice.T
+    return dslice
+
+
+def plot_parameters(what_to_plot, plot_name, diag_step=None):
```

Comment:
Let's move this into the field diagnostic class/file. It doesn't really make sense outside of it that I can tell. Please also add docstring.